modules/apps/commerce/commerce-pricing-service/src/main/java/com/liferay/commerce/pricing/internal/security/permission/resource/CommercePriceModifierRelModelResourcePermission.java

```diff
@@ -0,0 +1,121 @@
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	enabled = false, immediate = true,
+	property = "model.class.name=com.liferay.commerce.pricing.model.CommercePriceModifierRel",
+	service = ModelResourcePermission.class
+)
+public class CommercePriceModifierRelModelResourcePermission
+	implements ModelResourcePermission<CommercePriceModifierRel> {
+
+	@Override
+	public void check(
+			PermissionChecker permissionChecker,
+			CommercePriceModifierRel commercePriceModifierRel, String actionId)
+		throws PortalException {
+
+		CommercePriceModifier commercePriceModifier =
+			commercePriceModifierRel.getCommercePriceModifier();
+
+		commercePriceModifierRelPermission.check(
+			permissionChecker, commercePriceModifier.getCommercePriceListId(),
+			actionId);
+	}
+
+	@Override
+	public void check(
+			PermissionChecker permissionChecker,
+			long commercePriceModifierRelId, String actionId)
+		throws PortalException {
+
+		CommercePriceModifierRel commercePriceModifierRel =
+			commercePriceModifierRelLocalService.getCommercePriceModifierRel(
+				commercePriceModifierRelId);
+
+		CommercePriceModifier commercePriceModifier =
+			commercePriceModifierRel.getCommercePriceModifier();
+
+		commercePriceModifierRelPermission.check(
+			permissionChecker, commercePriceModifier.getCommercePriceListId(),
+			actionId);
+	}
+
+	@Override
+	public boolean contains(
+			PermissionChecker permissionChecker,
+			CommercePriceModifierRel commercePriceModifierRel, String actionId)
+		throws PortalException {
+
+		CommercePriceModifier commercePriceModifier =
+			commercePriceModifierRel.getCommercePriceModifier();
+
+		return commercePriceModifierRelPermission.contains(
+			permissionChecker, commercePriceModifier.getCommercePriceListId(),
+			actionId);
+	}
+
+	@Override
+	public boolean contains(
+			PermissionChecker permissionChecker,
+			long commercePriceModifierRelId, String actionId)
+		throws PortalException {
+
+		CommercePriceModifierRel commercePriceModifierRel =
+			commercePriceModifierRelLocalService.getCommercePriceModifierRel(
+				commercePriceModifierRelId);
+
+		CommercePriceModifier commercePriceModifier =
+			commercePriceModifierRel.getCommercePriceModifier();
+
+		return commercePriceModifierRelPermission.contains(
+			permissionChecker, commercePriceModifier.getCommercePriceListId(),
+			actionId);
+	}
+
+	@Override
+	public String getModelName() {
+		return CommercePriceModifierRel.class.getName();
+	}
+
+	@Override
+	public PortletResourcePermission getPortletResourcePermission() {
+		return null;
+	}
+
+	@Reference
+	protected CommercePriceModifierRelLocalService
+		commercePriceModifierRelLocalService;
+
+	@Reference
+	protected CommercePriceListPermission commercePriceModifierRelPermission;
```

Comment:
```suggestion
	protected CommercePriceListPermission commercePriceListPermission;
```

Reply:
Thank you @dsitu 
I knew that using "replace all", I would have miss something.